sampler.go

```diff
@@ -414,46 +408,34 @@ func NewRemotelyControlledSampler(
 ) *RemotelyControlledSampler {
 	initialSampler, _ := NewProbabilisticSampler(0.001)
 	sampler := &RemotelyControlledSampler{
```

Comment:
similar to reporter, can we create the final struct when all fields are available?